cudax/examples/stf/09-dot-reduce.cu

```diff
@@ -39,13 +39,11 @@ int main()
   auto lX = ctx.logical_data(X);
   auto lY = ctx.logical_data(Y);
 
-  auto lsum = ctx.logical_data(shape_of<scalar<double>>());
-
   /* Compute sum(x_i * y_i)*/
-  ctx.parallel_for(lY.shape(), lX.read(), lY.read(), lsum.reduce(reducer::sum<double>{}))
-      ->*[] __device__(size_t i, auto dX, auto dY, double& sum) {
-            sum += dX(i) * dY(i);
-          };
+  auto lsum = ctx.reduce(lY.shape(), reducer::sum<double>{}, lX.read(), lY.read())
+                ->*[] __device__(size_t i, auto dX, auto dY, double& sum) {
+                      sum += dX(i) * dY(i);
+                    };
```

Comment:
As a new user I'd be surprised that the order in which the order of parameters to `reduce` is different from the order of parameters to the lambda. I'd expect preservation of order (and perhaps restrictions, e.g. all reducers should be at the end):

```cpp
    auto lsum = ctx.reduce(lY.shape(), lX.read(), lY.read(), reducer::sum<double>{})
                ->*[] __device__(size_t i, auto dX, auto dY, double& sum) {
                      sum += dX(i) * dY(i);
                    };
````

or:

```cpp
    auto lsum = ctx.reduce(reducer::sum<double>{}, lY.shape(), lX.read(), lY.read())
                ->*[] __device__(size_t i, double& sum, auto dX, auto dY) {
                      sum += dX(i) * dY(i);
                    };
````

Reply:
My intuition was that the first is the better, with the op at the end, but I could not match it yet, perhaps we need to check that the op is not based on task_dep